### PR TITLE
Rewrite/homepage cleanup

### PR DESCRIPTION
**Pull request by tomvoelker**
## Intent

- Add REST API v2 endpoints and supporting DTOs/services/controllers:
  - Auth: POST /api/v2/auth/login, POST /api/v2/auth/logout, GET /api/v2/auth/me.
  - Users: GET /api/v2/users/{username}, GET /api/v2/users/{username}/tags, POST /api/v2/users (registration).
  - Tags: GET /api/v2/tags/{tagName} (with related tags).
  - Posts: POST /api/v2/posts, PUT /api/v2/posts/{postId}, DELETE /api/v2/posts/{postId} (owner-only).
  - Search: GET /api/v2/search (posts, users, tags).
  - Groups: GET /api/v2/groups, GET /api/v2/groups/{groupName} (with post counts).
- Implement corresponding services (AuthService, UserService, TagService, PostService, SearchService, GroupService), new DTOs (auth, user, post create/update, tag details, group DTOs, SearchResultDto), and controllers.
- Tighten SecurityConfig to explicitly permit login/logout and allow specific public read endpoints (instead of blanket /api/v2/auth/**).
- Frontend/homepage polish: layout/breakpoint adjustments (md → xl), remove page container borders, add Login CTA, convert tag cloud to nav with aria-label/title, change post rendering (clean bookmark URL, icons), and replace emoji placeholders.
- Add integration tests and test scaffolding for Auth, Tags, and Users controllers using stubbed LogicInterfaceFactory; update docs including next_apis.md and legacy-homepage analysis; change .env.development (remote API, mocks disabled), and minor i18n tweaks.

## Main Risks

- Authentication & Authorization
  - PII leakage: public read endpoints and conditional email exposure rely on correct request-scoped auth resolution—bugs could expose emails or other private fields.
  - Open registration risk: POST /api/v2/users is permitAll with no anti-abuse controls (CAPTCHA/rate-limits/verification), enabling spam/bot accounts.
  - Token/logout semantics: login returns API key token with expiresAt=null and logout is a no-op for stateless Basic Auth—clients may expect revocation/expiry semantics that are not implemented.
  - Security policy changes: tightening SecurityConfig may break clients relying on previous /api/v2/auth/** behavior.

- Data Integrity & Error Handling
  - Silent degradation: several services (TagService, UserService, GroupService, SearchService) log failures and return empty/partial results rather than clear errors—can hide server issues and create inconsistent client-visible data.
  - Validation and uniqueness: registration and some input validation appear minimal; lack of enforced uniqueness or stronger validation risks duplicate or malformed user records.
  - Ownership/enforcement: create/update/delete post flows must correctly enforce ownership; subtle logic errors could permit unauthorized modifications.

- Backward Compatibility & Developer Experience
  - Frontend .env and mocks: .env.development now points to a remote API and disables mocks—local development/test workflows may break.
  - API surface growth: many new DTOs and endpoints increase maintenance burden and require thorough documentation/migration notes.
  - Tests use detailed stubs: heavy reliance on stubbed LogicInterfaceFactory in tests may mask integration mismatches with production logic.

- UX & Content
  - Frontend rendering changes (bookmark description removal, URL presentation, icon changes) alter displayed content and could surprise users or downstream consumers.
  - Duplicate documentation insertion suggests accidental edit that needs cleanup.

## Follow-up Work

- Security & Auth
  - Implement token lifecycle (expiry/refresh) or document explicit token semantics; consider true token-based auth with revocation for logout.
  - Harden registration: add CAPTCHA, rate-limiting, email verification, and stricter username/email uniqueness checks.
  - Audit public endpoints and PII exposure; restrict email/groups by default or require explicit consent.

- Error Handling & Observability
  - Replace silent fallbacks with deterministic error responses or explicit partial-result metadata; surface meaningful HTTP codes for failure modes.
  - Improve logging, add metrics and Sentry-style error tracking for degraded paths.

- Tests & QA
  - Add negative/error-case integration tests (404/409/400 scenarios, auth edge cases, duplicate registration) and validate stub factories against production LogicInterface behavior.
  - Add tests verifying ownership enforcement for post updates/deletes.

- Back-compat & Documentation
  - Document SecurityConfig changes and migration guidance for existing clients.
  - Update next_apis.md with clear stability guarantees, rate-limit expectations, and PII rules.
  - Provide local-dev guidance (alternate .env or re-enable mocks) to avoid disrupting contributors.

- Codebase Cleanup
  - Remove duplicated content in HomePage.md and standardize i18n keys/casing.
  - Consider unifying test mock generation where it reduces surprise without increasing coupling (per contributor note).

### CODE REVIEW

File: bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/dto/AuthDto.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ * Returns the currently authenticated user's profile information.
+ */
+data class CurrentUserDto(
+    val name: String,
+    val realName: String?,
+    val email: String?,
+    val groups: List<String>,
+    val apiKey: String?
+)
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/dto/AuthDto.kt:53]: These DTOs serve different endpoints (/auth/login returns AuthUserDto, /auth/me returns CurrentUserDto) and may need to diverge in the future. The minor duplication is acceptable as a trade-off for flexibility. Keeping them separate also makes the API clearer about which DTO is used where.

File: bibsonomy-webapp-v2/.env.development [hunk 2]
```diff
@@ -1,10 +1,11 @@
 # Development environment configuration
 
 # API Configuration
-VITE_API_BASE_URL=http://localhost:8080
+# Use deployed backend on Kubernetes
+VITE_API_BASE_URL=http://lsx-kubemaster-1.informatik.uni-wuerzburg.de:30627/api/v2
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-webapp-v2/.env.development:5]: The external K8s endpoint is intentionally configured for integration testing against the deployed backend. For local development, developers can override this via local .env.local file or set VITE_ENABLE_MOCKS=true. This setup allows the CI preview deployments to work correctly against the staging API.

File: bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/service/UserService.kt [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        }
+        if (request.password.length < 8) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Password must be at least 8 characters")
+        }
+        if (request.email.isBlank()) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Email is required")
+        }
+        if (!request.email.contains("@")) {
+            throw ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid email address")
+        }
```
[the remark this answers is not in the document]
Comment (tomvoelker) [reply bibsonomy-rest-api-v2/src/main/kotlin/org/bibsonomy/api/service/UserService.kt:206]: The current validation is intentionally basic - a more robust regex or @Email annotation can be added later if needed. The service-layer validation provides a clear error message and catches the most common mistakes. Over-validating email addresses often leads to rejecting valid addresses.

**Comment by tomvoelker** [comment]
Regarding CodeRabbit feedback:

**Missing error scenario tests (line 74)**: These tests (404 for non-existing user, 409 for conflicts, 400 for validation) are valuable suggestions but out of scope for this initial API implementation PR. As noted in the test comments, these scenarios require more sophisticated mocking where the stub factory needs to behave differently based on input. Will address in follow-up.

**Inconsistent mock tag generation (line 328)**: The slight differences in mock tag generation (usercount formula) between test files is intentional - each test file has its own context and the specific values aren't meaningful beyond being plausible test data. Unifying would add coupling between tests without benefit.
